quarkus/config-api/src/main/java/org/keycloak/config/OptionBuilder.java

```diff
@@ -111,12 +114,32 @@ public List<String> get() {
         return this;
     }
 
+    public OptionBuilder<T> deprecated() {
+        this.deprecatedMetadata = new DeprecatedMetadata();
+        return this;
+    }
+
+    public OptionBuilder<T> deprecatedWithNote(String note) {
```

Comment:
Isn't enough the `deprecatedWithNote(String, String...)` method?

Also can we just name it `deprecated`? I mean, `WithNote` seems redundant as the parameter name is kinda making clear about setting a `note`.

To make easier to set deprecation without a note but the new options we just use a `deprecate(Set<Option>)`. So you have:

```
builder.deprecate(Set.of(OptionA, OptionB));
```

Reply:
Done.